clients/src/main/java/org/apache/kafka/clients/consumer/internals/OffsetFetcher.java

```diff
@@ -148,19 +149,8 @@ public Map<TopicPartition, OffsetAndTimestamp> offsetsForTimes(Map<TopicPartitio
             Map<TopicPartition, ListOffsetData> fetchedOffsets = fetchOffsetsByTimes(timestampsToSearch,
                     timer, true).fetchedOffsets;
 
-            HashMap<TopicPartition, OffsetAndTimestamp> offsetsByTimes = new HashMap<>(timestampsToSearch.size());
-            for (Map.Entry<TopicPartition, Long> entry : timestampsToSearch.entrySet())
-                offsetsByTimes.put(entry.getKey(), null);
-
-            for (Map.Entry<TopicPartition, ListOffsetData> entry : fetchedOffsets.entrySet()) {
-                // 'entry.getValue().timestamp' will not be null since we are guaranteed
-                // to work with a v1 (or later) ListOffset request
-                ListOffsetData offsetData = entry.getValue();
-                offsetsByTimes.put(entry.getKey(), new OffsetAndTimestamp(offsetData.offset, offsetData.timestamp,
-                        offsetData.leaderEpoch));
-            }
-
-            return offsetsByTimes;
+            return offsetFetcherUtils.buildOffsetsForTimesResult(timestampsToSearch,
```

Comment:
stylistically, can in put both in the same line? (I think it is less than 120char)

Reply:
Totally, definitely better looking